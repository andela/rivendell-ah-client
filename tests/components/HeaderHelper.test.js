import { shallow, mount } from 'enzyme';
import React from 'react';
import { Icon } from 'semantic-ui-react';
import { Router } from 'react-router-dom';
import {
  Logo, SearchBar, NavProfileImage, DesktopNavDropDown,
  DesktopNav, HamburgerMenu, CloseSideMenuButton, MobileNav
} from '../../src/components/helpers/HeaderHelper';

describe('Testing the Logo component', () => {
  it('Should render as expected', () => {
    const tree = shallow(<Logo />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the SearchBar component', () => {
  it('Should render as expected', () => {
    const tree = shallow(<SearchBar />);
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the NavProfileImage component', () => {
  it('Should render a profile image if it is passed', () => {
    const image = 'my-image-url';
    const profileImgClick = () => {};
    const tree = shallow(<NavProfileImage
      image={image} profileImgClick={profileImgClick}
    />);
    expect(tree.equals(<img
      alt="avatar"
      className="img"
      src={image}
      onClick={profileImgClick}
    />))
  });
  it('Should render a profile image icon if profile image is not passed', () => {
    const image = '';
    const profileImgClick = () => {};
    const profileImage = NavProfileImage({ image, profileImgClick });
    const tree = shallow(profileImage);
    expect(tree.equals(<i
      className="material-icons profile-img"
      onClick={profileImgClick}
    >
    account_circle
    </i>))
  });
});

describe('Testing the DesktopNavDropDown component', () => {
  const firstName = 'firstName';
  const logout = () => {};
  const username = 'username';
  let showDropDown = true;
  it('Should render if showDropDown is true', () => {
    const tree = shallow(<DesktopNavDropDown
      firstName={firstName}
      logout={logout}
      username={username}
      showDropDown={showDropDown}
    />)
    expect(tree).toBeDefined;
  });
  it('Should not render if showDropDown is true', () => {
    showDropDown = false
    const tree = shallow(<DesktopNavDropDown
      firstName={firstName}
      logout={logout}
      username={username}
      showDropDown={showDropDown}
    />)
    expect(tree).toBeNull;
  });
});

describe('Testing the DesktopNav component', () => {
  const firstName = 'firstName';
  const logout = () => {};
  const username = 'username';
  let token = '';
  const showDropDown = false;
  const image = 'my-image-url';
  const profileImgClick = () => {};
  const history = {
    location: {
      pathname: '/'
    },
    listen: () => {},
    createHref: () => {},
    push: () => {},
    replace: () => {}
  };  it('Should render a login button when no token is present', () => {
    const tree = mount(<Router history={history}><DesktopNav
      firstName={firstName}
      logout={logout}
      username={username}
      token={token}
      image={image}
      profileImgClick={profileImgClick}
      showDropDown={showDropDown}
    /></Router>);
    expect(tree.contains('Login')).toEqual(true);
  });
  it('Should render the notification icon when token is present', () => {
    token = 'token';
    const tree = mount(<Router history={history}><DesktopNav
      firstName={firstName}
      logout={logout}
      username={username}
      token={token}
      image={image}
      profileImgClick={profileImgClick}
      showDropDown={showDropDown}
    /></Router>);
    expect(tree.contains(<i className="material-icons">notifications_none</i>)).toEqual(true);
  });
});

describe('Testing the HamburgerMenu component', () => {
  it('Should render as expected', () => {
    const slideIn = () => {}
    const tree = shallow(<HamburgerMenu slideIn={slideIn} />)
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the CloseSideMenuButton component', () => {
  it('Should render as expected', () => {
    const slideOut = () => {}
    const tree = shallow(<CloseSideMenuButton slideOut={slideOut} />)
    expect(tree).toMatchSnapshot();
  });
});

describe('Testing the MobileNav component', () => {
  const firstName = 'firstName';
  const logout = () => {};
  const username = 'username';
  let token = '';
  const sideMenuStyles = {
    animationName: ''
  };
  const image = 'my-image-url';
  const profileImgClick = () => {};
  const slideOut = () => {}
  const history = {
    location: {
      pathname: '/'
    },
    listen: () => {},
    createHref: () => {},
    push: () => {},
    replace: () => {}
  };
  it('Should render a login button when no token is present', () => {
    const tree = mount(<Router history={history}><MobileNav
      firstName={firstName}
      logout={logout}
      username={username}
      token={token}
      image={image}
      profileImgClick={profileImgClick}
      slideOut={slideOut}
      sideMenuStyles={sideMenuStyles}
    /></Router>);
    expect(tree.contains(
      <li>
        <Icon name="sign-in" />
        Login
      </li>
    )).toEqual(true);
  });
  it('Should render the first name when token is present', () => {
    token = 'token';
    const tree = mount(<Router history={history}><MobileNav
      firstName={firstName}
      logout={logout}
      username={username}
      token={token}
      image={image}
      profileImgClick={profileImgClick}
      slideOut={slideOut}
      sideMenuStyles={sideMenuStyles}
    /></Router>);
    expect(tree.contains(<span className="first-name">{firstName}</span>)).toEqual(true);
  });
});
