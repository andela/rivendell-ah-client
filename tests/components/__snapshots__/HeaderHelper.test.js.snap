// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CloseSideMenuButton component Should render as expected 1`] = `
<div
  className="close-menu"
>
  <i
    className="material-icons"
    onClick={[Function]}
  >
    close
  </i>
</div>
`;

exports[`Testing the HamburgerMenu component Should render as expected 1`] = `
<div
  className="mobile hamburger-menu"
>
  <i
    className="material-icons"
    onClick={[Function]}
  >
    menu
  </i>
</div>
`;

exports[`Testing the Logo component Should render as expected 1`] = `
<div
  className="logo"
>
  <Link
    replace={false}
    to="/"
  >
    <svg
      fill="none"
      viewBox="0 0 129 80"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M64.5 0L129 80H95.0999L78.6047 60.1212H45.8921L62.3872 80H0L30.7449 41.8669L41.0642 54.303H73.7769L46.8657 21.8718L64.5 0Z"
        fill="#3498DB"
        fillRule="evenodd"
      />
    </svg>
  </Link>
</div>
`;

exports[`Testing the SearchBar component Should render as expected 1`] = `
<div
  className="search-box"
>
  <Form
    as="form"
  >
    <FormField>
      <input
        placeholder="search articles"
        type="search"
      />
    </FormField>
    <Button
      as="button"
      role="button"
    >
      <i
        className="material-icons"
      >
        search
      </i>
    </Button>
  </Form>
</div>
`;
